### PR TITLE
fixed-channelname-change-issue

### PR DESCRIPTION
**Pull request by Prateek93a**
@RocketChat/ReactNative

Closes #1283  #1169

Due to the same previous mistake, #1283 got messed up too. So I am raising this new PR with all the required changes and fixes.

### CODE REVIEW

File: app/views/RoomView/index.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ class RoomView extends React.Component {
 	static navigationOptions = ({ navigation }) => {
 		const rid = navigation.getParam('rid');
 		const prid = navigation.getParam('prid');
-		const title = navigation.getParam('name');
+		const title = navigation.state.params.title || navigation.getParam('name');
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: You not need to set two different params.
You can use only `navigation.getParam('name');`

File: app/views/RoomView/index.js [hunk 2]
```diff
@@ -336,6 +336,10 @@ class RoomView extends React.Component {
 		const observable = room.observe();
 		this.subSubscription = observable
 			.subscribe((changes) => {
+				const { navigation } = this.props;
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js]: Is better if you use only `name` param, like:
```
const name = navigation.getParam('name');
if (changes.name !== name) {
  navigation.setParams({ name: changes.name });
}
```

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Sorry for this long delay. Thanks for reviewing my code. Actually I thought initially the name parameter might not have any value so I used two variables with OR. But as you mentioned that in line 339 that it is better to use only name parameter instead of setting title, so the issue i was thinking will not arise. Thanks for this review Sir.

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Sorry Sir, I messed things up here. I will raise new PR for the issue. Sorry for the incovinience

**Comment by Prateek93a** [comment]
@djorkaeffalexandre Sorry Sir, I did not update the branch earlier, and also forgot to remove the changes I made for myself. But I have fixed everything now.

File: app/views/RoomView/index.js [hunk 3]
```diff
@@ -390,6 +390,11 @@ class RoomView extends React.Component {
...
+				const name = navigation.getParam('name');
+				if (changes.name !== name) {
+					navigation.setParams({ name: changes.name });
```
Comment (diegolmello) [inline app/views/RoomView/index.js]: I think we're missing other title headers here.
If it's a discussion or Real Name is enabled, the header comes from `fname`.
Check it out: https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/views/RoomsListView/index.js#L484

File: app/views/RoomView/index.js (hunk 2 above)
Comment (Prateek93a) [reply app/views/RoomView/index.js]: Ok Sir, made the changes accordingly

File: app/views/RoomView/index.js (hunk 1 above)
Comment (Prateek93a) [reply app/views/RoomView/index.js]: Thanks for the review Sir. I have made the changes.

File: app/views/RoomView/index.js (hunk 3 above)
Comment (Prateek93a) [reply app/views/RoomView/index.js]: Thanks for the review Sir. So first I should check whether name param is having some value or not, if there is no value, then I should use fname param, am I right on this?

**Comment by djorkaeffalexandre** [comment]
Can't reproduce on latest version.
Closing.
Thanks.

**Pull request closed without merging**